plugins/datastreamer_plugin/ui/topic_selection_dialog/Configuration.cpp

```diff
@@ -0,0 +1,106 @@
+        QDomDocument& doc,
+        QDomElement& parent_element) const
+{
+    DEBUG("Saving XML configuration state");
+
+    QDomElement max_elem = doc.createElement(MAX_ARRAY_SIZE_SETTINGS_TAG);
+    max_elem.setAttribute("value", QString::number(data_type_configuration.max_array_size));
+    parent_element.appendChild(max_elem);
+
+    QDomElement stamp_elem = doc.createElement(USE_HEADER_STAMP_SETTINGS_TAG);
+    stamp_elem.setAttribute("value", data_type_configuration.use_header_stamp ? "true" : "false");
+    parent_element.appendChild(stamp_elem);
+
+    QDomElement discard_elem = doc.createElement(DISCARD_LARGE_ARRAYS_SETTINGS_TAG);
+    discard_elem.setAttribute("value", data_type_configuration.discard_large_arrays ? "true" : "false");
+    parent_element.appendChild(discard_elem);
+
+    return true;
+}
+
+bool Configuration::xmlLoadState(
+        const QDomElement& parent_element)
+{
+    DEBUG("Loading XML configuration state");
+
+    QDomElement max_elem = parent_element.firstChildElement(MAX_ARRAY_SIZE_SETTINGS_TAG);
+    data_type_configuration.max_array_size = max_elem.attribute("value").toInt();
+
+    QDomElement stamp_elem = parent_element.firstChildElement(USE_HEADER_STAMP_SETTINGS_TAG);
+    data_type_configuration.use_header_stamp = (stamp_elem.attribute("value") == "true");
+
+    QDomElement discard_elem = parent_element.firstChildElement(DISCARD_LARGE_ARRAYS_SETTINGS_TAG);
+    data_type_configuration.discard_large_arrays = (discard_elem.attribute("value") == "true");
+
+    return true;
+}
+
+void Configuration::save_default_settings(
+        const QString& prefix) const
+{
+    // Get default Qt settings
+    QSettings settings;
+
+    settings.setValue(prefix + MAX_ARRAY_SIZE_SETTINGS_TAG, data_type_configuration.max_array_size);
+    settings.setValue(prefix + USE_HEADER_STAMP_SETTINGS_TAG, data_type_configuration.use_header_stamp);
+    settings.setValue(prefix + DISCARD_LARGE_ARRAYS_SETTINGS_TAG, data_type_configuration.discard_large_arrays);
+    settings.setValue(prefix + XML_DATATYPE_FILES_SETTINGS_TAG, xml_datatypes_files);
+    settings.setValue(prefix + DOMAIN_ID_SETTINGS_TAG, domain_id);
+}
+
+void Configuration::load_default_settings(
+        const QString& prefix)
+{
+    // Get default Qt settings
+    QSettings settings;
+
+    data_type_configuration.max_array_size = settings.value(prefix + MAX_ARRAY_SIZE_SETTINGS_TAG, data_type_configuration.max_array_size).toInt();
+    data_type_configuration.use_header_stamp = settings.value(prefix + USE_HEADER_STAMP_SETTINGS_TAG, data_type_configuration.use_header_stamp).toBool();
+    data_type_configuration.discard_large_arrays = settings.value(prefix + DISCARD_LARGE_ARRAYS_SETTINGS_TAG, data_type_configuration.discard_large_arrays).toBool();
+    xml_datatypes_files = settings.value(prefix + XML_DATATYPE_FILES_SETTINGS_TAG, xml_datatypes_files).toStringList();
+    domain_id = settings.value(prefix + DOMAIN_ID_SETTINGS_TAG, domain_id).toInt();
```

Comment:
Some clarification is needed here to know the effects of this function and where `data_type_configuration` is defined